### PR TITLE
[WIP] ApiClient

### PR DESCRIPTION
**Pull request by brianswko**
WIP steps towards using an instance-level connection

### CODE REVIEW

**Comment by brianswko** [comment]
@cfurrow these few changes make it work properly for `Trips.all` - which is enough for our current purposes - but fixing it across the entire gem is going to be a larger refactor which will fork this gem off into an entirely different direction.

**Comment by cfurrow** [comment]
That's probably fine for now, but the end-goal would be to propose these changes to the gem author so there aren't competing versions of this gem out there, or one that we're keeping just for ourselves.

File: lib/automatic/api_client/trips.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+module Automatic
+  class ApiClient
+    def trips
+      Trips.new(self)
+    end
+
+    class Trips
```
Comment (cfurrow) [inline lib/automatic/api_client/trips.rb:7]: This is a duplication of `Models::Trips`?

Comment (brianswko) [inline lib/automatic/api_client/trips.rb:7]: Essentially yeah - this was the initial way I addressed the issue that cropped up in https://github.com/lumoslabs/automatic-client/pull/2 before trying out the `method_missing` stuff. This took the class-level methods in `Models::Trips` and created duplicated instance-level methods but would need some pretty major DRYing out to be usable across the other model classes and methods

**Pull request closed without merging**